mdiag/mdiag.sh

```diff
@@ -832,24 +830,19 @@ function _ungraboutput {
 
 function _json_strings_arrayify {
 	local a=("$@")
-	a=("${a[@]//\\/\\\\}") # this fixes vim syntax highlighting -> "
-	a=("${a[@]//\"/\\\"}")
-	a=("${a[@]//	/\\t}")
-	a=("${a[@]//$_lf/\\r}")
-	a=("${a[@]/#/\"}")
-	a=("${a[@]/%/\",}")
-	if [ ${#a[@]} -gt 0 ]; then
-		a[$(( ${#a[@]} - 1 ))]="${a[$(( ${#a[@]} - 1 ))]%,}"
-	fi
+	for i in ${!a[@]}; do
+	   a[$i]=$(_json_stringify "${a[$i]}")
```

Comment:
I bet you were cursing me while figuring out how the old code worked...  :)

As below, the motivation was to avoid forking, whereas this way forks once per line (to run `_json_stringify`).  The code clarity benefit is very large, though.

Since the updated `_json_stringify` is also forking a bit, have you run the old and new mdiag one after the other on the same system, and timed how long each takes to run (overall)?  I suspect that the performance hit is negligible, but it might be good to check.

Reply:
Having this code path call `_json_stringify` only slowed down mdiag.sh runs on my test machine by 5 seconds at most. As you say, I think the clarity benefit outweighs the slight hit.